lib/ansible/module_utils/kubevirt.py

```diff
@@ -134,6 +134,14 @@ def merge_dicts(base_dict, merging_dicts):
         merging_dicts can be a dict or a list or tuple of dicts.  In the latter case, the
         dictionaries at the front of the list have higher precedence over the ones at the end.
         """
+        def _deepupdate(D, E):
+            for k in E:
+                if isinstance(E[k], dict) and isinstance(D.get(k), dict):
+                    D[k] = _deepupdate(D[k], E[k])
+                else:
+                    D[k] = E[k]
+            return D
```

Comment:
I think you're overcomplicating and obfuscating this code.  If you need recursive merging, add this loop to merge_dicts() rather than creating a new, nested function with obscured variable names.

Or, an even better option: talk with @sdoran and @sivel about moving ansible.utils.vars.merge_hash() somewhere in ansible.module_utils.common (perhaps ansible.module_utils.common.collections) and use that directly.

Reply:
@abadger my immediate goal here is to remove the regression that 2.8.1 introduced. Being able to use more of ansible's code is a great idea, and I hope @samdoran and @sivel follow your suggestion, so I can use that on devel, but for now I need something that is quickly mergeable.

Ideally it would be this exact version, as otherwise I'll have to write three versions of this code:
- the one currently on devel
- the new one on devel once `merge_hash()` is available
- and a completely new one, not used anywhere else, just for stable-2.8, just so I can fix a regression.

How long before 2.8.2 is released?

Reply:
2.8.2 will be released Thursday, next week.  Last day to merge backports for it is on Monday when I go home for the day.

Yeah, please fix this code as I mentioned above.  You may be fixing the regression but you're doing it by re-introducing the obfuscation that was gotten rid of.  You won't need to maintain three versions.  Fix the code in devel so that the loop is a part of merge_dicts() as I specified and then add that commit to this backport.  That will be the only one until merge_hash() is available.

Reply:
Added a commit that drops the nested function, hope that's what you had in mind.

Reply:
I was looking into moving `ansible.utils.vars.merge_hash()` and discovered we have `ansible.module_utils.common.dict_transformations.dict_merge()` that was added in #41471. I wonder if that function could be used in this case.

Reply:
@samdoran yup, it's a drop–in replacement for the nested function Toshio wanted dropped.

@abadger modified my commit, now the nested function got replaced with the ansible–provided one. Unit tests pass, so it should work fine.

Reply:
@mmazur, cool, do you have a pull request for devel as well?  I can merge both today.

Reply:
@abadger https://github.com/ansible/ansible/pull/58286